tools/integration_tests/rapid_appends/rapid_appends_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rapid_appends
+
+import (
+	"log"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// Number of appends to perform on test file.
+const numAppends = 3
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+
+// TODO: Split the suite in two suites single mount and multi-mount.
+type RapidAppendsSuite struct {
+	suite.Suite
+	fileName    string
+	fileContent string
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func (t *RapidAppendsSuite) SetupSuite() {
+	setup.MountGCSFuseWithGivenMountFunc(flags, mountFunc)
+	secondaryMntTestDirPath = setup.SetupTestDirectory(testDirName)
+}
+
+func (t *RapidAppendsSuite) TearDownSuite() {
+	setup.UnmountGCSFuse(secondaryMntRootDir)
+	if t.T().Failed() {
+		log.Println("Secondary mount log file:")
+		setup.SaveGCSFuseLogFileInCaseOfFailure(t.T())
+		log.Println("Primary mount log file:")
+		setup.SetLogFile(primaryMntLogFilePath)
+		setup.SaveGCSFuseLogFileInCaseOfFailure(t.T())
+	}
+}
+func (t *RapidAppendsSuite) createUnfinalizedObject() {
+	t.fileName = client.FileName1 + setup.GenerateRandomString(5)
+	// Create unfinalized object.
+	t.fileContent = setup.GenerateRandomString(client.SizeOfFileContents)
+	_ = client.CreateUnfinalizedObject(ctx, t.T(), storageClient, path.Join(testDirName, t.fileName), t.fileContent)
```

Comment:
nit: I don't think we need to assign to _ here.

```suggestion
	client.CreateUnfinalizedObject(ctx, t.T(), storageClient, path.Join(testDirName, t.fileName), t.fileContent)
```